### PR TITLE
Catch TypeErrors in hazard.base.Hazard.write_hdf5

### PR DESCRIPTION
**Pull request by emanuel-schmid**
E.g., `climada.hazard.Wildfire` has class members that are not readily suitable for being written to an hdf5 file.<br>
This leads to a `TypeError` being raised when one tries to write the data down in this format.

To avoid this and grant the minimal functionality, i.e., creating a file that can be read again into a `Hazard` object, the writing has now an Error tolerance.

Note that this still means that a `Wildfire` object cannot be recreated completely with the `read_hdf5` method. However, this is generally the case and how to deal with it needs to be addressed in a separate thread.

The PR solves issue #383.

### CODE REVIEW

**Comment by peanutfun** [comment]
I'll pick this up as my first PR to review, so please be patient with me 😅

**Comment by peanutfun** [comment]
@emanuel-schmid Thank you for this efficient fix! At least for the `Hazard`  base class, everything works fine. I added a unit test to check that.

@carmensteinmann can you confirm that this resolves your issue? Storing the file should now result in warnings for the attributes `FirmsParams` and `ProbaParams`, and loading the file should create a `Wildfire` object that holds these two attributes with their default values.

@chahank I added a unit test to explicitly check for the new error handling. Could you quickly check if this is in line with the developer's guide? 🙃

**Comment by emanuel-schmid** [comment]
@peanutfun : adding a unit test: excellent! 🙌

File: climada/hazard/test/test_base.py [hunk 1]
```diff
@@ -1142,6 +1142,32 @@ def test_write_read_pass(self):
...
+        # Create a hazard with unsupported type as attribute
+        hazard = dummy_hazard()
+        hazard.event_id = CustomID()
+
+        # Write the hazard and check the logs for the correct warning
+        with self.assertLogs(logger="climada.hazard.base", level="WARN") as cm:
+            hazard.write_hdf5(file_name)
+        self.assertRegex(
+            cm.output[0], r"write_hdf5: the class member event_id is skipped"
+        )
```
[the remark this answers is not in the document]
Comment (peanutfun) [reply climada/hazard/test/test_base.py:1162]: Absolutely! I missed that assertion, apparently.

**Comment by carmensteinmann** [comment]
@peanutfun I ran the code with an actual wildfire hazard and it works as you described - thanks a lot. 

The only thing I'm wondering is whether there is another solution than setting FirmsParams and ProbaParams to the default values when reading the hazard. Maybe for now we could add the information that this is the case to the warning message, so at least the person saving the hazard is aware of it? (Would it be complicated to add that warning when reading the hazard as well?)

**Comment by peanutfun** [comment]
@carmensteinmann Great to hear that it works ☺️  You now suggested two extensions of the warnings, right?

1. Extend the warning when *writing*, stating that skipped variables will be set to default values when loading. That's easy, I'll do that!
2. Issue a warning when a class member is not found when *reading* from an H5 file. The respective code is [`climada/hazard/base.py#L1216`](https://github.com/CLIMADA-project/climada_python/blob/ebcaddec6f020030e19347c8fe85d3254bd3b7ab/climada/hazard/base.py#L1216). I'm not that familiar with Python class inheritance. If I call `Wildfire.from_hdf5()`, is `cls` an instance of `Wildfire` in that case? Then it should be easy to implement because the method is indeed aware of all members of `Wildfire`. I'll give it a try.

> The only thing I'm wondering is whether there is another solution than setting FirmsParams and ProbaParams to the default values when reading the hazard.

The (complicated) solution is that every class derived from `Hazard` defining data types that are unsupported by `h5py` must provide its own read/write method.

**Comment by carmensteinmann** [comment]
> You now suggested two extensions of the warnings, right?

Yes, exactly. I think this would be a great first implementation,  so the user has this issue in mind. I'm also not familiar with class inheritance, maybe @emanuel-schmid can help us out here?

The more complicated extension might be part of a larger discussion in the future on how to handle saving exceptions in general.

**Comment by carmensteinmann** [comment]
> This does not work well, unfortunately. The base class method is not aware of the members of a derived class, and additionally, other members like `pool` are not written but appear in the warnings when reading. I guess we need to skip that for now

I think that's alright - thanks for trying it out @peanutfun!

**Review by carmensteinmann — APPROVED** [review]

**Review by peanutfun — COMMENTED** [review]
Good catch! I though the f-string would apply to the following lines as well 🙈

**Review by peanutfun — APPROVED** [review]

**Pull request merged**